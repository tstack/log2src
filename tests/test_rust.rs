use assert_cmd::prelude::*;
use std::{path::Path, process::Command};
use insta_cmd::assert_cmd_snapshot;

fn get_platform_suffix() -> &'static str {
    if cfg!(target_os = "windows") {
        "windows"
    } else {
        "unix"
    }
}

#[test]
fn basic() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("log2src")?;
    let source = Path::new("examples").join("basic.rs");
    let log = Path::new("tests")
        .join("resources")
        .join("rust")
        .join("basic.log");
    cmd.arg("-d")
        .arg(source.to_str().expect("test case path is valid"))
        .arg("-l")
        .arg(log.to_str().expect("test case log path is valid"))
        .arg("-f")
        .arg(r#"\[\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z \w+ \w+\]\s+(?<body>.*)"#);

    let snapshot_name = format!("basic_{}", get_platform_suffix());
    assert_cmd_snapshot!(snapshot_name, cmd);

    Ok(())
}

#[test]
fn stack() -> Result<(), Box<dyn std::error::Error>> {
    let mut cmd = Command::cargo_bin("log2src")?;
    let source = Path::new("examples").join("stack.rs");
    let log = Path::new("tests")
        .join("resources")
        .join("rust")
        .join("stack.log");
    cmd.arg("-d")
        .arg(source.to_str().expect("test case path is valid"))
        .arg("-l")
        .arg(log.to_str().expect("test case log path is valid"))
        .arg("-f")
        .arg(r#"\[\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}Z \w+ \w+\]\s+(?<body>.*)"#)
        .arg("-s")
        .arg("1");

    let snapshot_name = format!("stack_{}", get_platform_suffix());
    assert_cmd_snapshot!(snapshot_name, cmd);
    Ok(())
}
