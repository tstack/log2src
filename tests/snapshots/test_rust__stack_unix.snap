---
source: tests/test_rust.rs
info:
  program: log2src
  args:
    - "-d"
    - examples/stack.rs
    - "-l"
    - tests/resources/rust/stack.log
    - "-f"
    - "\\[\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}Z \\w+ \\w+\\]\\s+(?<body>.*)"
    - "-s"
    - "1"
---
success: true
exit_code: 0
----- stdout -----
{"srcRef":{"sourcePath":"examples/stack.rs","lineNumber":15,"column":11,"name":"b","text":"\"Hello from b\"","args":[],"vars":[]},"variables":{},"stack":[[{"sourcePath":"examples/stack.rs","lineNumber":11,"column":4,"name":"a","text":"b","args":[],"vars":[]},{"sourcePath":"examples/stack.rs","lineNumber":7,"column":4,"name":"main","text":"a","args":[],"vars":[]}]]}

----- stderr -----
